mitodl/mitxpro
# apply coupons to all course runs of a course

#### Pre-Flight checklist


#### What are the relevant tickets?
https://trello.com/c/TLujv7yA/57-apply-coupons-to-all-courseruns-of-a-course

#### What's this PR do?

- Added new field in coupon`include_future_runs`
- Include new filed in django admin and coupon creation form
- New Django signal will add another CouponEligibility object to all such Coupons with `include_future_runs=True` that are already eligible for other runs of the same course.

#### How should this be manually tested?

- Create a new coupon for a course run with `include_futurer_runs=True`,
- Create another course run product. 
- Coupon should be available for newly created course run.

#### Screenshots (if appropriate)
<img width="812" alt="Screenshot 2020-03-04 at 5 17 34 PM" src="https://user-images.githubusercontent.com/4245618/75879176-c1299d00-5e3c-11ea-92a8-0b53c46729ea.png">

![Screenshot 2020-03-04 at 5 18 58 PM](https://user-images.githubusercontent.com/4245618/75879204-cc7cc880-5e3c-11ea-8c13-f79166fe0dae.png)
